lib/ddtrace/context.rb

```diff
@@ -16,319 +13,48 @@ module Datadog
-        # In other cases, this is just broken and one should rely
-        # on per-instrumentation code to handle parent/child relations.
-        self.current_span_op = parent
-
-        # If root span has been closed and spans are still unfinished...
-        # ...emit some warnings/metrics to bring attention to this.
-        # All spans should be closed when the root span closes.
-        # Otherwise traces can leak, or be associated incorrectly.
-        if parent.nil? && !all_spans_finished?
-          if Datadog.configuration.diagnostics.debug
-            opened_spans = @trace.length - @finished_spans
-            Datadog.logger.debug("Root span #{span_op.name} closed but has #{opened_spans} unfinished spans:")
-          end
-
-          @trace.reject(&:finished?).group_by(&:name).each do |unfinished_span_name, unfinished_spans|
-            Datadog.logger.debug("Unfinished span: #{unfinished_spans.first}") if Datadog.configuration.diagnostics.debug
-            Datadog.health_metrics.error_unfinished_spans(
-              unfinished_spans.length,
-              tags: ["name:#{unfinished_span_name}"]
-            )
-          end
-        end
-      end
-    end
-
-    # Returns if the trace for the current Context is finished or not. A \Context
-    # is considered finished if all spans in this context are finished.
-    def finished?
-      @mutex.synchronize do
-        return all_spans_finished?
-      end
-    end
-
-    # @@return [Numeric] numbers of finished spans
-    def finished_span_count
-      @mutex.synchronize do
-        @finished_spans
-      end
-    end
-
-    # Returns true if the context is sampled, that is, if it should be kept
-    # and sent to the trace agent.
-    def sampled?
-      @mutex.synchronize do
-        return @sampled
-      end
-    end
-
-    # Returns both the trace list generated in the current context and
-    # if the context is sampled or not.
-    #
-    # It returns +[nil,@sampled]+ if the \Context is
-    # not finished.
-    #
-    # If a trace is returned, the \Context will be reset so that it
-    # can be re-used immediately.
-    #
-    # This operation is thread-safe.
-    #
-    # @return [Array<Array<Span>, Boolean>] finished trace and sampled flag
-    def get
-      @mutex.synchronize do
-        # Return sampled attribute, even if context is not finished
-        sampled = @sampled
-        return nil, sampled unless all_spans_finished?
-
-        # Collect finished spans
-        trace = @trace.collect { |op| op.send(:span) }
-
-        # Root span is finished at this point, we can configure it
-        annotate_for_flush!(trace.first)
-
-        # Allow caller to modify trace before context is reset
-        yield(trace) if block_given?
-
-        # Reset the context for re-use.
-        reset
-
-        # Return Span measurements and whether it was sampled.
-        [trace, sampled]
-      end
+    def initialize(
```

Comment:
Could you provide documentation for the public methods in this class?
Sorry if this was already planned for the future and I'm calling it out too early.

Reply:
This class still needs unit tests, but I don't intend for this to be public. All the public APIs that return `Context` objects have been eliminated, so this should be completely internal.

Reply:
To clarify: I'm looking for yard docs for the public methods in this class, for ourselves the devs to understand the methods.